sti/utils.lua

```diff
@@ -170,4 +170,45 @@ function utils.convert_isometric_to_screen(map, x, y)
 		(tileX + tileY) * tileH / 2
 end
 
+function utils.hexToColor(Hex)
+	
+	if Hex:sub(1, 1) == "#" then
+		
+		Hex = Hex:sub(2)
+		
+	end
+	
+	return tonumber( Hex:sub(1, 2), 16 ), tonumber( Hex:sub(3, 4), 16 ), tonumber( Hex:sub(5, 6), 16 )
+	
+end
+
+function utils.fixTransparentColor(tileset)
+	
+	if tileset.transparentcolor then
+	
+		local r, g, b = utils.hexToColor(tileset.transparentcolor)
+		local ImageData = tileset.image:getData()
+		
+		for x = 0, ImageData:getWidth() - 1 do
+			
+			for y = 0, ImageData:getHeight() - 1 do
+				
+				local pr, pg, pb = ImageData:getPixel(x, y)
+				
+				if r == pr and g == pg and b == pb then
+					
+					ImageData:setPixel(x, y, r, g, b, 0)
```

Comment:
Just a suggestion: this while loop could be written cleaner using [`ImageData:mapPixel`](https://love2d.org/wiki/ImageData:mapPixel).